policy/templates/subtemplates/auto/auto-test.gotmpl

```diff
@@ -205,6 +205,23 @@
 
       {{- end }} {{/* end of if .test */}}
 
+      - name: Upload Playwright Test Report to S3
+        if: failure() && steps.test_execution.outcome != 'success' && steps.env_up.outcome == 'success'
+        run:
+          npm run upload_report_to_s3
+        env:
+          AWS_ACCESS_KEY_ID: {{`${{ secrets.UI_AWS_ACCESS_KEY_ID }}`}}
+          AWS_SECRET_ACCESS_KEY: {{`${{ secrets.UI_AWS_SECRET_ACCESS_KEY }}`}}
+          RUN_ID: 'tyk-analytics/{{`${{ github.run_id }}`}}'
+        working-directory: tyk-analytics/tests/ui
```

Comment:
Isn't this report already handled in https://github.com/TykTechnologies/gromit/blob/master/policy/templates/subtemplates/auto/auto-test.gotmpl#L204 and https://github.com/TykTechnologies/gromit/blob/master/policy/templates/subtemplates/auto/auto-test.gotmpl#L230?

Reply:
No. Playwright generates html report (stored in tests/ui/playwright-report) and is essential when you need to analyze failed tests. Right now we do not store it anywhere and devs can't investigate failed tests.
I also don't think that everyone have access to this AWS account (for example I don't).

Reply:
ok. A few things:
1. Lets upload to one S3 bucket. The XML reports are uploaded to an S3 bucket that is controlled by us and is not public. A sub-directory can be made public for this.
2. Make the HTML report generation explicit in the playwright run step

Reply:
1. This bucket is also controlled by us, and it's a special bucket for UI tests results (ARA, portal, dash tests), with all subdirectories and retention rules. I would rather use that one
2. HTML report is generated during tests execution by default (this is how playwright was designed), no extra implementation needed here

Reply:
@alephnull I plan to merge it tomorrow (this is critical feature for our pipelines). Are you OK with that?